New Wrap Secret

jlm0 opened the pull request:
**Updates for Wrap Secret:**
- Completely new UI. 
- Most of the logic has stayed constant but has been split up into smaller project files with appropriate names.
- Chains that need to be suggested (like terra) have their own object in common.ts.
- Chains and Token list in config have the majority of up-to-date data. Still missing some data from the excel sheet.
- Types have been moved over to their own page types.ts
- Transactions have been split up relative to their component. Data is now retrieved based on the active token component.
- Toast notification on the catch of most errors.
- The UI background changes colors based on the active token. Colors have been preset based on tokens in the list.
**Notes**
When adding a new chain the `config.ts`, `types.ts` and the `tokenIcons` object in `"./assets/images/index.ts"` all should be updated. If the chain needs to be suggested then the `SuggestedChains` object in commons needs to be updated. Right now that's Terra Classic and Chihuahua but will need to include Terra 2.

**Future Possible Improvements**
- Better (more specific) error handling.
- Consistency on promise handling.
- Validate user field inputs.
- Fallback LCD/RPC endpoints.
- Testnet Wrapping/Bridging Support.

Commit:
setup project

Commit:
Setup Vite Ts project

Commit:
feat: connected fonts. Added images, favicon, colors, theming.

Commit:
feat: prepared components: Background, Accordion.

Commit:
feat: responsive wrap page UI

Commit:
feat: added missing libs, rebase files: config, commons

Commit:
fix: unify token types

Commit:
ref: rebase keplr wallet connection

Commit:
fix: static mainnet network

Commit:
feat: disable coming soon tokens, start token rows rebasing

jlm0 committed:
content

update content wording and links

Commit:
rebase: almost all token rows logic

Commit:
feat: calculate balance percentage

Commit:
fix:
 - rebase connect wallet action from header to button,
 - disable bridge tab if no viewing key (+animation),
 - remove connect wallet from wrap/unwrap elems

Commit:
feat: add market capitalization, price change

jlm0 committed:
setKeplr in header

Header connect wallet should set Keplr as well.

jlm0 committed:
split to helpers

Split Content and TokenForm components into helper functions to make code more readable and easier to track.

jlm0 committed:
update indicator component

Add interval updates to data. Remove on component unmount. Format strings.

jlm0 committed:
Merge branch 'master' into rebase-logic

jlm0 committed:
Update Balances

Restructure getting balances. Moved requests into helpers. Requesting data for balances occurs on component mounting individually and every few intervals.

jlm0 committed:
Balance in USD

Format and display balance in USD.

jlm0 committed:
Viewkey Handler

Handle setting view key if there is none.

jlm0 committed:
Loader and change props

Add Loader when getting market, price, and or balance data. Update props to have token rather than tokenOptions so the current token isn't searched for over and over using tokenOptions.

jlm0 committed:
point to scrt.network

point to scrt.network rpc and lcd endpoints

jlm0 committed:
Merge branch 'master' into rebase-logic

jlm0 committed:
update assets

update static assets and split up config from types

jlm0 committed:
Wrap/Unwrap

Update wrap and unwrap functions.

jlm0 committed:
IBC Bridge

Update logic for deposit/withdrawal over ibc. Also updated configuration files with up to date information from relayers.
-ToDo: Catch errors and fix stylings.

jlm0 committed:
Merge branch 'master' into rebase-logic

jlm0 committed:
Error Handling and Toast Notification

Add error handling for data retrieval and Txs. Provide user feedback for error/success via toast notification.

jlm0 committed:
Final Updates

Move components up 1 file level.
Add a carousel for tokens since the list will get long.
Clean up stylings.
Edit some error handling.

jlm0 committed:
Merge branch 'master' into rebase-logic

jlm0 commented:
Assaf, please review and let me know of any suggested improvements. I think the majority of bugs were caught. A few styling updates can be done for better responsiveness.

assafmo commented:
How can I run this locally? Both `npm i && npm run start` and `yarn && yarn start` output this error:
```
[plugin:vite:import-analysis] Failed to resolve import "./components/Background/Background" from "src/App.tsx". Does the file exist?
/home/assafmo/wrap.scrt.network/src/App.tsx:3:0
15 |  var _jsxFileName = "/home/assafmo/wrap.scrt.network/src/App.tsx", _s = $RefreshSig$();
16 |  import { useState, useEffect } from "react";
17 |  import { Background } from "./components/Background/Background";
   |                              ^
18 |  import { Content } from "./components/Content/Content";
19 |  import { TokenNames } from "./types";
```

jlm0 committed:
Merge remote-tracking branch 'upstream/master'

jlm0 committed:
Fix Types

Fix types and import to allow sucesful local build.

jlm0 committed:
Rename Background

Locally the file was correct but each push had a Camel case name which causes vite problem when cloning the remote repo.

jlm0 committed:
Update config

jlm0 committed:
secp error

secp buffer error on latest keplr version. reverting back.

jlm0 committed:
update config

update endpoints

jlm0 committed:
Add global event listener

add listener for keystorechange

jlm0 committed:
fee

fix feedenom

jlm0 committed:
Merge remote-tracking branch 'upstream/master'

jlm0 committed:
Token List Component and Chains

Add LUNA to suggested chains. Change swiper carousel navigation and add stylings.

jlm0 committed:
update stylings

update stylings and cleanup some types

jlm0 committed:
Opengraph and Docs

Add links to FAQ to new docs and update opengraph.

jlm0 committed:
..

jlm0 committed:
Revert ".."

This reverts commit 9038349792f7c8ab7f3b649b5b1c674bd4ff8ce8.

jlm0 committed:
Minor Fixes

Fix images. On Window event dont force reconnect of wallet. Add current token to wrap input field to match bridge. Clean up some stylings.

jlm0 committed:
Manual Refresh

Add refresh button to manually refresh balances.

jlm0 committed:
Update terra RPC

jlm0 committed:
remove log

cleanup console logs

jlm0 committed:
suggestChain

Add suggestChain to targetchain on secret deposit/withdraw.

jlm0 committed:
Update grav.svg

jlm0 committed:
update opengraph
